collectors/proc.plugin/proc_net_wireless.c

```diff
@@ -0,0 +1,452 @@
+    for (current = netwireless_root; current;) {
+
+        if (timercmp(&current->updated, timestamp, <)) {
+            struct netwireless *to_free = current;
+            current = current->next;
+            netwireless_free(to_free);
+
+            if (previous) {
+                previous->next = current;
+            } else {
+                netwireless_root = current;
+            }
+        } else {
+            previous = current;
+            current = current->next;
+        }
+    }
+}
+
+// finds an existing interface or creates a new entry
+static struct netwireless *find_or_create_wireless(const char *name)
+{
+    struct netwireless *wireless;
+    uint32_t hash = simple_hash(name);
+
+    // search it, from begining to the end
+    for (wireless = netwireless_root ; wireless ; wireless = wireless->next) {
+        if (unlikely(hash == wireless->hash && !strcmp(name, wireless->name))) {
+            return wireless;
+        }
+    }
+
+    // create a new one
+    wireless = callocz(1, sizeof(struct netwireless));
+    wireless->name = strdupz(name);
+    wireless->hash = hash;
+
+    // link it to the end
+    if (netwireless_root) {
+        struct netwireless *last_node;
+        for (last_node = netwireless_root; last_node->next ; last_node = last_node->next);
+
+        last_node->next = wireless;
+    } else
+        netwireless_root = wireless;
+
+    return wireless;
+}
+
+static void configure_device(int do_status, int do_quality, int do_discarded_packets, int do_missed,
+                             struct netwireless *wireless_dev) {
+    wireless_dev->do_status = do_status;
+    wireless_dev->do_quality = do_quality;
+    wireless_dev->do_discarded_packets = do_discarded_packets;
+    wireless_dev->do_missed_beacon = do_missed;
+    wireless_dev->configured = true;
+
+    char buffer[RRD_ID_LENGTH_MAX + 1];
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_status", wireless_dev->name);
+    wireless_dev->chart_id_net_status = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_link", wireless_dev->name);
+    wireless_dev->chart_id_net_link = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_level", wireless_dev->name);
+    wireless_dev->chart_id_net_level = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_noise", wireless_dev->name);
+    wireless_dev->chart_id_net_noise = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_discarded_packets", wireless_dev->name);
+    wireless_dev->chart_id_net_discarded_packets = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_missed_beacon", wireless_dev->name);
+    wireless_dev->chart_id_net_missed_beacon = strdupz(buffer);
+}
+
+int do_proc_net_wireless(int update_every, usec_t dt)
+{
+    UNUSED(dt);
+    static procfile *ff = NULL;
+    static int do_status, do_quality = -1, do_discarded_packets, do_beacon;
+    static char *proc_net_wireless_filename = NULL;
+
+    if (unlikely(do_quality == -1)) {
```

Comment:
you read configuration in this branch, why do we check `do_quality`?

Reply:
This is only a simple verification that the collector was already initialized. It could be any of the static variables.

Reply:
Correct me if i am wrong, you read(init) config in that branch? If so why not `init_config` var?

Reply:
Ilya the name here is only a detail, this is a common code that we have inside [health](https://github.com/netdata/netdata/blob/a0900ae81151211322160043d564141f8f9584fe/health/health_config.c#L506) and other places of the code. We need a variable that was not initialized and we are sure that after it to be initialized all other variables that need initialization will be initialized.

Reply:
i understood, still looks hacky imo. @vlvkobal what do you think?